### PR TITLE
pelican-bootstrap3: refactored sidebar (take 2)

### PR DESCRIPTION
**Pull request by digitalrounin**
This is more of a proposal for refactor of the sidebar.  Figured it
would be easier to do and discuss in a PR than it would be an
issue/ticket.

These changes can be split out if only some of the ideas are accepted.
It is mostly a lot cleanup and shuffling around to make it a bit more
consistent.

Here is a summary of the changes:

  - Moved all sidebar related from `includes/` and into
    `includes/sidebar/`.
  - Moved the following sections in `includes/sidebar.html` into
    their own files under `includes/sidebar/`:
    - _Social_ -> `includes/sidebar/social.html`
    - _Recent Posts_ -> `includes/sidebar/recent_posts.html`
    - _Categories_ -> `includes/sidebar/categories.html`
    - _Tags_ -> `includes/sidebar/tag_cloud.html`
    - _Series_ -> `includes/sidebar/series.html`
  - A whole bunch of whitespace cleanup.
  - Added the `DISABLE_SIDEBAR_TITLE_ICONS` configuration option. If
    `DISABLE_SIDEBAR_TITLE_ICONS` is set to True, none of the sidebar
    title icons will be displayed.
  - Folded in from getpelican/pelican-themes#438: Added the
    `SIDEBAR_ON_LEFT` configuration option, which moves the sidebar
    to the left side of the page.  Sidebar will continue to appear on
    the bottom of the page on smaller screens.
  - Assuming getpelican/pelican#2072 worked out adding a new `THEME_TEMPLATE_OVERRIDES` configuration setting:
    - Optional include of `includes/sidebar/optional_top.html` and
      `includes/sidebar/optional_bottom.html` from within
      `includes/sidebar.html`.  This would allow a user to insert
      sections into the sidebar.
    - `includes/sidebar/recent_posts.html` include articles line items
      via `includes/sidebar/article-li.html`.  This allows customization.

### CODE REVIEW

**Comment by digitalrounin** [comment]
@justinmayer,
Continuation from getpelican/pelican-themes#440.  Apologies for this mix up.

So I made the changes, and while testing, with an updated version of getpelican/pelican, and ran into the _pelican-bootstrap3: UndefinedError: 'gettext' is undefined_ (getpelican/pelican-themes#460) issue.

I ran out of time and revered my work on the original `sidebar_refactor` branch, that's where I closed the original PR.

Either way, new work is in the `sidebar_refactor+merge_update` branch, waiting for getpelican/pelican-themes#460 to get resolved.  I will try and hack at this again in a week or two.

**Comment by digitalrounin** [comment]
@justinmayer,
Please disregard my previous comment in this PR.  Conflicts with `master` resolved, and this is ready for review.

I read more into getpelican/pelican-themes#460 and saw that I missed the `PLUGINS = ['i18n_subsites']` from getpelican/pelican-themes#465.

Let me know if there are any other changes required.

**Comment by digitalrounin** [comment]
@jranke,
Could you please review if you have the time?  This overlaps with changes you made.


Cheers!

File: pelican-bootstrap3/templates/includes/sidebar/links.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{% if LINKS %}
+  {% from 'includes/sidebar/macros.jinja' import title %}
+
+<!-- Sidebar/Links -->
+<li class="list-group-item">
+  <h4>{{ title('Links', DISABLE_SIDEBAR_TITLE_ICONS, icon='external-link-square') }}</h4>
```
Comment (jranke) [inline pelican-bootstrap3/templates/includes/sidebar/links.html]: Here, 'Links' should be replaced by _('Links') so this can be translated, too.

**Comment by jranke** [comment]
@digitalrounin I spotted one string that should be made i18n-aware, see above. As far as I can tell from just looking at the code, the PR looks fine to me.

**Comment by digitalrounin** [comment]
Good catch @jranke!  I corrected the issue.

Ready to go if it looks good to you @justinmayer?

**Comment by almet** [comment]
Great, thanks for taking the time to submit this (and sorry for the delay on accepting the changes :-))

**Comment by digitalrounin** [comment]
Yay!  Thanks so much @almet.

Any chance of taking a peek at getpelican/pelican#2072, please?  It allows for theme template overloading.

Cheers!

**Pull request merged**